src/core/shape_inference/include/slice_shape_inference_utils.hpp

```diff
@@ -163,6 +163,29 @@ TDim make_dim(const TDim& dim, const Bounds& start, const Bounds& stop, int64_t
 
     return {lb, ub};
 }
+
+/**
+ * \brief Computes the size of the default strides ([1, 1, ..., 1]) of a StridedSlice.
+ *
+ * \param begin  Begin argument of a StridedSlice.
+ * \param end    End argument of a StridedSlice.
+ *
+ * \return The static size of the strides or a negative value if the size is dynamic.
+ */
+inline int64_t default_stride_size(const Output<Node>& begin, const Output<Node>& end) {
```

Comment:
Change begin, end to be template shape, same type as input shapes in shape_infer function.